File: src/main/java/com/azilzor/chatterchain/service/UserService.java
package com.azilzor.chatterchain.service;

import java.util.LinkedHashSet;
import java.util.Set;

import org.springframework.stereotype.Service;

import com.azilzor.chatterchain.dto.UserDto;

@Service
public class UserService {

  private final Set<UserDto> onlineUsers = new LinkedHashSet<>();

  public Set<UserDto> getOnlineUsers() {
    return onlineUsers;
  }

  public void addUser(UserDto user) {
    onlineUsers.add(user);
  }

  public void removeUser(UserDto user) {
    onlineUsers.remove(user);
  }

}

File: src/main/kotlin/com/azilzor/chatterchain/service/UserService.kt
package com.azilzor.chatterchain.service

import com.azilzor.chatterchain.dto.UserDto
import org.springframework.stereotype.Service

@Service
class UserService {
    private val onlineUsers: MutableSet<UserDto?> = LinkedHashSet()

    fun getOnlineUsers(): Set<UserDto?> {
        return onlineUsers
    }

    fun addUser(user: UserDto?) {
        onlineUsers.add(user)
    }

    fun removeUser(user: UserDto?) {
        onlineUsers.remove(user)
    }
}
